internal/handlers/common/update.go

```diff
@@ -622,6 +698,57 @@ func extractValueFromUpdateOperator(op string, update *types.Document) (*types.D
 		return nil, NewWriteErrorMsg(ErrFailedToParse, "Modifiers operate on fields but we found another type instead")
 	}
 
+	if op == "$rename" {
```

Comment:
We should probably update the function description or move `$rename` checks into a separate function.

Reply:
I've placed it in this function, because it also checks for duplicates afterwards and returns the error about path collisions, I'll add a comment about checking for collisions inside of it.

Btw, thanks to your catch I guess we found a bug in duplicates handling!

Reply:
I agree with Dmitry. We added a cyclomatic complexity with this `if`. I'm ok to call some function in this place, but I'd suggest moving the logic inside this `if` into a separate function, so the code is more readable.

Reply:
Moved the code to seperate `validateRenameExpression` function